Dockerfile

```diff
@@ -0,0 +1,93 @@
+FROM centos:7
+ENV container docker
+
+# Set ENV, LANG only needed if building with docker-1.8
+ENV LANG en_US.UTF-8
+ENV TERM xterm
+
+# Fetch postgresql 9.4 COPR repo
+RUN curl -sSLko /etc/yum.repos.d/rhscl-rh-postgresql94-epel-7.repo \
+https://copr-fe.cloud.fedoraproject.org/coprs/rhscl/rh-postgresql94/repo/epel-7/rhscl-rh-postgresql94-epel-7.repo
+
+## Install EPEL repo, yum necessary packages for the build without docs, clean all caches
+RUN yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && \
+    yum -y install --setopt=tsflags=nodocs \
+                   bison                   \
+                   bzip2                   \
+                   cmake                   \
+                   file                    \
+                   gcc-c++                 \
+                   git                     \
+                   libffi-devel            \
+                   libtool                 \
+                   libxml2-devel           \
+                   libxslt-devel           \
+                   libyaml-devel           \
+                   make                    \
+                   memcached               \
+                   net-tools               \
+                   nodejs                  \
+                   openssl-devel           \
+                   patch                   \
+                   rh-postgresql94-postgresql-server \
+                   rh-postgresql94-postgresql-devel  \
+                   readline-devel          \
+                   sqlite-devel            \
+                   which                   \
+                   &&                      \
+    yum clean all
+
+# Add persistent data volume for postgres
+VOLUME [ "/var/opt/rh/rh-postgresql94/lib/pgsql/data" ]
+
+# Download chruby and chruby-install, install, setup environment, clean all
+RUN curl -sL https://github.com/postmodern/chruby/archive/v0.3.9.tar.gz | tar xz && \
+    cd chruby-0.3.9 && make install && scripts/setup.sh && \
+    echo "gem: --no-ri --no-rdoc --no-document" > ~/.gemrc && \
+    echo "source /usr/local/share/chruby/chruby.sh" >> ~/.bashrc && \ 
+    curl -sL https://github.com/postmodern/ruby-install/archive/v0.6.0.tar.gz | tar xz && \
+    cd ruby-install-0.6.0 && make install && ruby-install ruby 2.2.4 -- --disable-install-doc && \
+    echo "chruby ruby-2.2.4" >> ~/.bash_profile && \
+    rm -rf ~/ruby* && rm -rf /usr/local/src/* && yum clean all
+
+## Environment for scripts
+RUN echo "export BASEDIR=/manageiq" > /etc/default/evm && \
```

Comment:
Not sure I understand the question here.

The ENV variables are required by various scripts not during build time, hence the use of this file.